### ISSUE TITLE
Bad dependency checks behaviour

### ISSUE THREAD

**Issue by dshkol** [mountainMath/cancensus#180]
Testing out installing the package on a completely clean M1 Air. Installed `cancensus` and tested examples without previously installing any spatial packages.

```
library(cancensus)
> census_data <- get_census(dataset='CA16', regions=list(CMA="59933"),
+                           vectors=c("v_CA16_408","v_CA16_409","v_CA16_410"),
+                           level='CSD', geo_format = "sf")
Error in get_census(dataset = "CA16", regions = list(CMA = "59933"), vectors = c("v_CA16_408",  : 
  the `sf` package is required to return geographies.
```
A call with a spatial format should first check whether required packages are installed and prompt installation instead of straight to error. 

After installing `sf` tried again:
```
> library(cancensus)
> census_data <- get_census(dataset='CA16', regions=list(CMA="59933"),
+                           vectors=c("v_CA16_408","v_CA16_409","v_CA16_410"),
+                           level='CSD', geo_format = "sf")
Querying CensusMapper API...
Querying CensusMapper API...                                                                                        
Downloading: 52 kB     Error in loadNamespace(x) : there is no package called ‘geojsonsf’
```
Similarly, this should be tested and caught before attempting to query and therefore throwing an error.

**Comment by mountainMath** [comment]
Alternatively we could just require these two packages. Realistically speaking, who is going to use the package purely for tabular data without geographies, and be worried about having to install `sf` and `geojsonsf`?

**Comment by dshkol** [comment]
Yeah, but sf has a lot of overhead to install and I'd rather avoid making it a hard dependency and leave it to the user to enable if they don't have it and want spatial data. I'll add that to a to-do list.

**Comment by mountainMath** [comment]
I guess it can get annoying to install sf on a Mac or Linux because one might have to install libraries? Forgot.

So the idea is to still fail but more gracefully, telling people more nicely that they have to install sf to use the specific function?

**Comment by dshkol** [comment]
> So the idea is to still fail but more gracefully, telling people more nicely that they have to install sf to use the specific function?

I've seen other libraries prompt for an install of the suggested package when used in a function like that. Shouldn't break any workflows that already use that call with geospatial data enabled.

**Comment by mountainMath** [comment]
Maybe, but what if the code is not run in an interactive session?

**Comment by mountainMath** [comment]
Implemented in upcoming v0.5.3.
